app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsViewModel.kt

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.searchsuggestions
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.ViewModelProvider
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.Transformations.map
+import android.arch.lifecycle.ViewModel
+import android.content.Context
+import android.graphics.Typeface
+import android.text.Spannable
+import android.text.SpannableStringBuilder
+import android.text.style.StyleSpan
+import mozilla.components.browser.search.SearchEngine
+
+sealed class State {
+    data class Disabled(val givePrompt: Boolean) : State()
+    data class NoSuggestionsAPI(val givePrompt: Boolean) : State()
+    class ReadyForSuggestions : State()
+}
+
+class SearchSuggestionsViewModel(
+    private val fetcher: SearchSuggestionsFetcher,
+    private val preferences: SearchSuggestionsPreferences
+) : ViewModel() {
+    private val _selectedSearchSuggestion = MutableLiveData<String>()
+    val selectedSearchSuggestion: LiveData<String> = _selectedSearchSuggestion
```

Comment:
What I had before was a getter for the MutableLiveData that gave it as a LiveData. @mcomella recommended I switch it: https://github.com/boek/focus-android/commit/818c04737e32f7af30b172d0dea79458a043849c#r29939651

Reply:
Where is the setter? (maybe the code changed?)

There's a lot of boilerplate but I don't think there's much overhead: afaik, this would compile to two field references (not functions). I think it's good to separate mutability from immutability so I'd prefer not to expose MutableLiveData.

Probably not worth it if no one already knows how but I wonder if we could reduce the boilerplate by either writing an annotation or generating this code. I wonder what the android docs do?

Reply:
@mcomella all of the examples I've seen have a private MutableLiveData and then expose it with a `getLiveData(): LiveData` function.